src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/X500NameEncoder.ManagedDecode.cs

```diff
@@ -135,22 +138,18 @@ internal static partial class X500NameEncoder
             return decodedName.ToString();
         }
 
-        private static string ReadString(DerSequenceReader tavReader)
+        private static string ReadString(AsnReader tavReader)
         {
-            var tag = (DerSequenceReader.DerTag)tavReader.PeekTag();
+            var tag = tavReader.PeekTag();
 
-            switch (tag)
+            switch ((UniversalTagNumber)tag.TagValue)
             {
```

Comment:
This cast doesn't feel right.

Reply:
It's fine if you first check that the TagClass is Universal.